src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AlwaysEncryptedKeyConverter.cs

```diff
@@ -137,16 +102,16 @@ internal static byte[] ECDHPublicKeyToECCKeyBlob(ECDiffieHellmanPublicKey public
             byte[] keyBlob = new byte[ECCPublicKeyBlob.Size];
 
             // Set magic number
-            Array.Copy(KeyBlobMagicNumber.ECDHPublicP384, 0, keyBlob, 0, 4);
+            Buffer.BlockCopy(KeyBlobMagicNumber.ECDHPublicP384, 0, keyBlob, 0, 4);
             // Set key size
             keyBlob[4] = (byte)ECCPublicKeyBlob.KeySize;
 
             ECPoint ecPoint = publicKey.ExportParameters().Q;
             Debug.Assert(ecPoint.X.Length == ECCPublicKeyBlob.KeySize && ecPoint.Y.Length == ECCPublicKeyBlob.KeySize,
                $"ECDH public key was not the expected length. Actual (X): {ecPoint.X.Length}. Actual (Y): {ecPoint.Y.Length} Expected: {ECCPublicKeyBlob.Size}");
             // Copy x and y coordinates to key blob
-            Array.Copy(ecPoint.X, 0, keyBlob, ECCPublicKeyBlob.HeaderSize, ECCPublicKeyBlob.KeySize);
-            Array.Copy(ecPoint.Y, 0, keyBlob, ECCPublicKeyBlob.HeaderSize + ECCPublicKeyBlob.KeySize, ECCPublicKeyBlob.KeySize);
+            Buffer.BlockCopy(ecPoint.X, 0, keyBlob, ECCPublicKeyBlob.HeaderSize, ECCPublicKeyBlob.KeySize);
```

Comment:
From what I've read, using either one would make little difference in this situation. Any particular reason for changing it?

Reply:
Consistency with the rest of the library. We use blockcopy 41 times and array copy 24 times. As you say there is no performance advantage either way. If you prefer the original I'll change it back, let me know.

Reply:
Doesn't matter to me, just curious :grin: